mne/channels/channels.py

```diff
@@ -235,6 +237,84 @@ def get_channel_types(self, picks=None, unique=False, only_data_chs=False):
         return _get_channel_types(self.info, picks=picks, unique=unique,
                                   only_data_chs=only_data_chs)
 
+    def get_ch_positions(self, picks=None):
+        """Get a dictionary of channel positions for each channel.
+
+        Channel positions are in meters in the specified coordinate frame
+        of the montage. All channel positions are a 3-dimensional vector
+        corresponding to xyz positions in meters. For MEG channels, they
+        also add the vectorized orientation.
+
+        For more information on the different channel type
+        positions obtained, see :class:`mne.Info` documentation.
+
+        Parameters
+        ----------
+        picks : str | list | slice | None
+            None corresponds to all data channels that are not MEG.
+
+        Returns
+        -------
+        ch_positions : dict
+            A dictionary with keys as the ch_names and values as
+             corresponding list of xyz coordinates in the montage
+             coordinate frame.
+        ch_frame_dict : dict
+            A dictionary with keys as the ch_names and values as
+             corresponding montage coordinate frame.
+        """
+        # only get channels that were picked
+        picks = _picks_to_idx(self.info, picks)
+        chs = [self.info['chs'][ipick] for ipick in picks]
+
+        # extract name, position, orientation (for MEG)
+        ch_names, ch_pos = [], []
+        for ch in chs:
+            pos = ch['loc'][:3].copy()
+            ch_pos.append(pos)
+            ch_names.append(ch['ch_name'])
+
+        # extract landmark coords and coordinate frame
+        ch_coord_frame = [_frame_to_str[ch['coord_frame']] for ch in chs]
+
+        # construct channel position dictionary and coord frame
+        ch_pos_dict = dict(zip(ch_names, ch_pos))
+        ch_frame_dict = dict(zip(ch_names, ch_coord_frame))
+
+        return ch_pos_dict, ch_frame_dict
+
+    @fill_doc
+    def get_montage(self):
+        """Get a DigMontage from instance.
+
+        Returns
+        -------
+        %(montage)s
+        """
+        from ..channels.montage import make_dig_montage
+        if self.info['dig'] is None:
+            return None
+
+        # extract landmark coords and coordinate frame
+        landmark_coords, coord_frame_int = _get_fid_coords(self.info['dig'])
+        coord_frame = _frame_to_str[coord_frame_int]
+
+        # create montage and return it
+        montage_bunch = _get_data_as_dict_from_dig(self.info['dig'])
+        ch_names = self.info['ch_names']
+        ch_locs = montage_bunch.dig_ch_pos_location
+        ch_pos = dict(zip(ch_names, ch_locs))
+        montage = make_dig_montage(
+            ch_pos=ch_pos,
+            coord_frame=coord_frame,
+            nasion=montage_bunch.nasion,
+            lpa=montage_bunch.lpa,
+            rpa=montage_bunch.rpa,
+            hsp=montage_bunch.hsp,
+            hpi=montage_bunch.hpi,
+        )
```

Comment:
I am fine with this code that I trust but why having method get_ch_positions if you can just do:

raw.get_montage().ch_pos

to get channel positions?

Reply:
it's also not using the same code so it's error prone

Reply:
So rn, I'm not sure why but the latest commit fails when doing this. That is, i) I get a montage and its coordinates, then ii) I `raw.set_montage()`, then iii) I try to get the coordinates using either `chs['loc'][:3]`, or `.get_montage()._get_ch_pos()`.

In iii), the `chs['loc'][:3]` works fine, but for some reason `.get_montage()._get_ch_pos()` does not, and so I want to suspect that there is a transformation that occurs when doing `set_montage()` because the roundtrip of `get_montage` and `set_montage` works fine.

Reply:
I just pushed up the change and you'll see that the tests fail here:

```
mne/channels/tests/test_montage.py:1329: in test_get_ch_positions
    assert_array_equal(montage_ch_pos[ch], pos)
E   AssertionError: 
E   Arrays are not equal
E   
E   Mismatched elements: 3 / 3 (100%)
E   Max absolute difference: 0.06785223
E   Max relative difference: 0.78861487
E    x: array([-0.001064,  0.018188,  0.127724], dtype=float32)
E    y: array([-0.001133,  0.08604 ,  0.081454], dtype=float32)
```

Reply:
It's a ref channel problem, pushed a commit